tests/utils.py

```diff
@@ -48,19 +52,24 @@ def wrapped_condition_function():
             return condition_function(**kwargs)
         return condition_function()
 
-    if 'timeout' in kwargs:
-        timeout = kwargs['timeout']
-        del kwargs['timeout']
-    else:
-        timeout = TIMEOUT
-
     start_time = time.time()
     while time.time() < start_time + timeout:
-        if wrapped_condition_function():
+        condition_val = wrapped_condition_function()
+        if expected_value is None:
+            if condition_val:
+                return True
+        elif condition_val == expected_value:
             return True
         time.sleep(0.5)
 
-    raise WaitForTimeout(get_message())
+    if get_message:
+        message = get_message()
+    elif expected_value:
+        message = 'Final value: {}'.format(condition_val)
```

Comment:
Would have helped me debug `test_removing_component_while_its_getting_updated` - I thought the callback wasn't getting called at all, when in fact it was called too many times! With `expected_value` you can get error messages like:
```
======================================================================
ERROR: test_removing_component_while_its_getting_updated (tests.test_render.Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_render.py", line 1423, in test_removing_component_while_its_getting_updated
    wait_for(lambda: call_counts['button-output'].value, expected_value=1)
  File "tests/utils.py", line 72, in wait_for
    raise WaitForTimeout(message)
WaitForTimeout: Final value: 2
```
Which would have made it much clearer what was going on.

Would be nice if we could share utils like this across dash repos - perhaps if we split out `dash_development` things like this could go there?